### ISSUE TITLE
Allow selective installation of requirements via setuptools extras_require

### ISSUE THREAD

**Issue by petri** [opsdroid/opsdroid#766]
When more extensions are added to the opsdroid core distribution, the number of dependencies will increase, making the package larger and adding potential installation issues.

The setuptools extras_require mechanism can be used to make installation of dependencies optional, alleviating this issue.

**Comment by jacobtomlinson** [comment]
I agree. We should combine this with #651 as part of a dependency refactor.

_It is worth noting that only three dependencies have been added as a result of merging the eight database and connector modules into core._
